### PR TITLE
v1 readme

### PR DESCRIPTION
**Pull request by tonyzhu163**
draft 1 of markdown handout

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+> Collaboration: Individual milestone 
+> Completion: About 12–16 hours 
+> Deadline: Tuesday Apr. 9, 2024 (11:59PM ET) 
+> Latest handout version: CS1380:2024:M5 
+> GitHub repo: https\://github.com/brown-cs1380/m5
+
+The goal of this milestone is to implement a scalable programming model and abstractions for processing large data sets in a distributed fashion.The core of the execution engine combines two higher order functions: a map function, which performs filtering and sorting and a reduce function, which performs a summarization operation. These functions operate over the distributed storage system implemented in the previous milestone, by serializing the distributed functions, attaching and running various tasks in parallel, managing all communications and data transfers between the various parts of the system.
+
+## Table of Contents
+- Background & Context
```
Comment (nvasilakis) [inline docs/README.md:12]: These are not links to the document, are they?

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+all.store.get(null, (e, v) => {
+  all.mr.exec({keys: v, map: m1, reduce: r1}, console.log)
+});
+```
+
+We will see the details of how this implemented after we discuss the core abstractions, but at a high level, exec designates the current node as the coordinator orchestrating several phases of the computation, it sets up a set of new endpoints across various nodes, it sends the keys and functions to all nodes, and then signals the execution of  the MapReduce computation. On each node, the first phase operates on key-object pairs from the node's local storage system and once complete notifies the coordinator. After receiving a full set of completions from all nodes in the group, the coordinator notifies all nodes to move onto the reduction phase. As they complete this phase, nodes send results to the coordinator. When all results are gathered, the coordinator invokes the callback function.
+
+| Service    | Description | Methods |
+| -------- | ------- | ------- |
+| mr  | A MapReduce implementation    |   **exec**     |
```
Comment (nvasilakis) [inline docs/README.md:74]: `exec` with backquotes, not **exec** using bold:-)

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+Execute eslint frequently, and use the --fix flag to correct some of the style errors automatically.
+Test your implementation using your own MapReduce workflows!
+
+To create a submission, run s/submit.sh from the root folder of M4. This will create a submission.zip file which you will then upload to Gradescope (select "managed submission" and then upload your zip manually). The zip archive will contain a javascript/src/main folder inside which all the files tracked by git and found to be modified. Make sure to git commit any new files you have created before running the submit script; the s/submit.sh script will not include any untracked files in the submission.
+
+You are allowed to submit as many times as you want up until the deadline; so submit early and often. For full grade, before submitting the final version of your code make sure that (1) all linters run without any errors, (2) the provided tests run without any errors, and (3) you have provided an additional five or more tests with your implementation.
+```
```
Comment (nvasilakis) [inline docs/README.md:261]: We might want to add a FAQ section here.

**Review by nvasilakis — CHANGES_REQUESTED** [review]
This looks great! Minor changes requested, thanks!

**Comment by tonyzhu163** [comment]
@nvasilakis 

Resolved changes mentioned.

**Review by nvasilakis — APPROVED** [review]

**Pull request merged**